packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -997,3 +959,1336 @@ class _RenderTheatre extends RenderBox with ContainerRenderObjectMixin<RenderBox
+  /// in the widget tree (which is usually the enclosing [Route]).
+  ///
+  /// The built overlay child widget, is inserted below this widget in the
+  /// widget tree, allowing it to depend on [InheritedWidget]s above it, and
+  /// be notified when the [InheritedWidget]s change.
+  ///
+  /// Unlike [child], the built overlay child can visually extend outside the
+  /// bounds of this widget without being clipped, and receive hit-test events
+  /// outside of this widget's bounds, as long as it does not extend outside of
+  /// the [Overlay] on which it is rendered.
+  final WidgetBuilder overlayChildBuilder;
+
+  /// A widget below this widget in the tree.
+  final Widget? child;
+
+  @override
+  State<OverlayPortal> createState() => _OverlayPortalState();
+}
+
+// An _OverlayLocation whose occupants get painted above a specific
+// [OverlayEntry] (but below the [OverlayEntry] above that [OverlayEntry]).
+class _OverlayEntryLocation extends _OverlayLocation with LinkedListEntry<_OverlayEntryLocation> {
+  _OverlayEntryLocation(this._showTimestamp, this._theatreChildModel) : super._();
+
+  final int _showTimestamp;
+  final _EventOrderChildModel _theatreChildModel;
+  _RenderDeferredLayoutBox? _overlayChildRenderBox;
+
+  @override
+  _RenderTheatre get _theatre => _theatreChildModel._theatre;
+
+  @override
+  void _addToChildModel(_RenderDeferredLayoutBox child) {
+    assert(_overlayChildRenderBox == null, 'Failed to add $child. $_overlayChildRenderBox is attached to $this');
+    assert(_showTimestamp != null);
+    _overlayChildRenderBox = child;
+    _theatreChildModel._add(this);
+  }
+
+  @override
+  void _removeFromChildModel(_RenderDeferredLayoutBox child) {
+    assert(child == _overlayChildRenderBox);
+    _overlayChildRenderBox = null;
+    assert(_theatreChildModel._sortedChildren.contains(this));
+    _theatreChildModel._remove(this);
+  }
+
+  @override
+  void _activate(_RenderDeferredLayoutBox child) {
+    assert(_debugNotDisposed());
+    assert(_overlayChildRenderBox == null, '$_overlayChildRenderBox');
+    super._activate(child);
+    _overlayChildRenderBox = child;
+  }
+
+  @override
+  void _deactivate(_RenderDeferredLayoutBox child) {
+    assert(_debugNotDisposed());
+    super._deactivate(child);
+    _overlayChildRenderBox = null;
+  }
+}
+
+class _RenderTheatreMarker extends InheritedWidget {
+  const _RenderTheatreMarker({
+    required this.theatre,
+    required this.overlayEntryWidgetState,
+    required super.child,
+  });
+
+  final _RenderTheatre theatre;
+  final _OverlayEntryWidgetState overlayEntryWidgetState;
+
+  @override
+  bool updateShouldNotify(_RenderTheatreMarker oldWidget) {
+    return oldWidget.theatre != theatre
+        || oldWidget.overlayEntryWidgetState != overlayEntryWidgetState;
+  }
+
+  @override
+  InheritedElement createElement() => _RenderTheatreMarkerElement(this);
+
+  static _RenderTheatreMarker? of(BuildContext context, { required bool targetRootOverlay }) {
```

Comment:
as far as I can tell, targetRootOverlay is always set to false. Do we need this functionality?